dev/release_notes.readme.md

```diff
@@ -0,0 +1,102 @@
+
+We show the subset of those merged pull requests that belong to the release in question and are regarded as relevant for the release notes.  For each of these pull requests, one line is shown, containing the number (as a link to the pull request in github) and the title.  Pull requests with related topics are grouped together under suitable subheadings.  These groups are ordered according to decreasing "severity".  Of course it is not possible to define such a general linear ordering, but we want each pull request to appear **only once** in the release notes, not in several groups according to several aspects it belongs to.
+
+Note that a file with release notes that is expected to be **read** must be **short**; other tools would be needed for **evaluating** the release notes (searching/filtering facilities according to various aspects).
+
+Besides the list of relevant pull requests, the release notes contain also the lists of changed GAP packages:
+- omitted packages (which had been distributed with the previous GAP version but are not distributed with the current one),
+- changed packages (which are distributed with both the previous and the current GAP version, but with different package versions),
+- new packages (which had not been distributed with the previous GAP version but are distributed with the current one).
+
+For each package in question, there is one entry that shows package name (with a link to the package homepage), version, authors and abstract.
+
+Note that the information about packages cannot be extracted from pull requests. It depends on the archives of packages that are distributed with GAP.
+(It would be easy to put this information together if there woud be a text file for each archive that lists which package versions are contained in the archive.)
+
+
+## How are release notes created from the github pull requests?
+
+The main idea is to use the *labels* that are assigned to the pull requests.
+Currently the following labels are regarded as relevant for the creation of release notes.
+
+- Pull requests with the label `release notes: not needed` shall **not** be listed in the release notes.
+  Thus the release notes are composed from all those merged pull requests that belong to the release in question (are assigned to the release branch or have a `backport-to-...-DONE` label for the release) AND do NOT have the label `release notes: not needed`.
+
+- For pull requests with the label `release notes: use title`, the text for the release notes is given by the title of the pull request.
+  For all other pull requests, the text for the release notes can be extracted from the pull request body.
+  (The pull request template contains `## Text for release notes`, thus the relevant text is expected there; but where is the end of this text? We have to define a convention to mark the release notes text, then this text can be extracted automatically.)
```

Comment:
Exactly. At that time, it was not envisaged that the we will automatically retrieve and process the PR description. We should be able to do this now, but a marker for the end of the text will be desirable.

Reply:
Yes. I will extend the pull request template.